### PR TITLE
remove unused dependencies (notify-api-375)

### PR DESCRIPTION
**Pull request by terrazoon**
## Description

The deptry tool noted that the these removed dependencies are actually dependencies in notifications-utils, so we are not really removing most of them, just not declaring them, since they belong to notifications-utils.

The tool also says there are some implicit dependencies that ought to be explicitly declared, etc.

## Security Considerations

N/A

### CODE REVIEW

**Review by A-Shumway42 — APPROVED** [review]
Looks good to me. deptry tool seems really neat!

File: pyproject.toml [hunk 1]
```diff
@@ -9,44 +9,38 @@ readme = "README.md"
...
-cffi = "==1.16.0"
-charset-normalizer = "^3.1.0"
 click = "==8.1.7"
 click-datetime = "==0.2"
-click-didyoumean = "==0.3.0"
-click-plugins = "==1.1.1"
-click-repl = "==0.3.0"
-deprecated = "==1.2.14"
-eventlet = "==0.35.2"
+deptry = "==0.8.0"
```
Comment (ccostino) [inline pyproject.toml:18]: `deptry` should be a development dependency, right?  It's only used by us locally with the `make` command?

File: pyproject.toml [hunk 2]
```diff
@@ -9,44 +9,38 @@ readme = "README.md"
...
+eventlet = "==0.34.3"
```
Comment (ccostino) [inline pyproject.toml:19]: This is downgrading `eventlet` for some reason - it's marked at `0.35.2` above, and I know we recently updated it from Dependabot.

**Review by ccostino — CHANGES_REQUESTED** [review]
Thanks, @terrazoon!

This is handy for sure; is there a high degree of confidence of what this is removing vs. changing?  Would it make more sense to tackle this after seeing what we can do about eliminating notification-utils completely by consolidating into the admin and/or API repos?

If we feel this is still safe to do now, I did note a couple of things that I think to be adjusted before we can merge it.  Thanks!

**Comment by terrazoon** [comment]
> Thanks, @terrazoon!
> 
> This is handy for sure; is there a high degree of confidence of what this is removing vs. changing? Would it make more sense to tackle this after seeing what we can do about eliminating notification-utils completely by consolidating into the admin and/or API repos?
> 
> If we feel this is still safe to do now, I did note a couple of things that I think to be adjusted before we can merge it. Thanks!

Well, if we don't trust the tool we have to do it by hand.  Would by hand be more reliable?  The ticket is to remove unused dependencies and from what I've seen this tool finds them.

**Comment by ccostino** [comment]
We'll come back to this after the `ordered_set` and Python 3.12 update!

**Pull request closed without merging**